### ISSUE TITLE
Tutorial on https://vector.dev/guides/integrate/sources/syslog/loki/#start-vector failed

### ISSUE THREAD

**Issue by nadgobpx01** [timberio/vector#2776]
The tutorial on:

https://vector.dev/guides/integrate/sources/syslog/loki/#start-vector

Here's what went wrong:
Jun 10 14:45:32.435 ERROR vector: Configuration error: "vector.toml": missing field `encoding` for key `sinks.out`

**Comment by nadgobpx01** [comment]
adding this line fixed it:

 #Encoding
encoding.codec = "json" # optional, default

**Comment by binarylogic** [comment]
Thanks, going to reopen this so we can fix the docs.

**Comment by pySilver** [comment]
@nadgobpx01 have you tried this with their cloud offering?

This:

```
[sinks.loki]
  # General
  type = "loki"
  inputs = ["docker", "syslog"]
  endpoint = "https://logs-prod-us-central1.grafana.net"

  # Auth
  auth.user = "4902"
  auth.password = "<api token here>"
  auth.strategy = "basic"

  # Encoding
  encoding.codec = "json" # optional, default

  # Labels
  labels.job = "varlogs"
```

gives 404 error in vector sink log :(

**Comment by pySilver** [comment]
which is very strange since 

~ ❯ curl https://4902:therewasmytoken\=\=@logs-prod-us-central1.grafana.net                                                                                                                                                    8s 16:22:45
OK%

**Comment by nadgobpx01** [comment]
@pySilver 
somehow yesterday responded well on health check , on later try today i have the output  from issue #2778.Thank you .

**Comment by ktff** [comment]
Tutorials were correct, encoding should be optional. 

Fixed with #2796.

**Comment by hb0nes** [comment]
```
ii  vector         0.28.1-1     amd64        A lightweight and ultra-fast tool for building observability pipelines
```

Encoding does not seem optional. Examples error out on me due to encoding.
The examples don't specify encoding, but the docs do show this:
![afbeelding](https://user-images.githubusercontent.com/38855893/227178537-ce5eb450-8ee3-4915-a573-5849e6f0d797.png)
